core/blockchain.go

```diff
@@ -632,7 +681,7 @@ func (bc *BlockChain) setHeadBeyondRoot(head uint64, time uint64, root common.Ha
 					if root != (common.Hash{}) && !beyondRoot && newHeadBlock.Root() == root {
 						beyondRoot, rootNumber = true, newHeadBlock.NumberU64()
 					}
-					if !bc.HasState(newHeadBlock.Root()) {
+					if !bc.StateRecoverable(newHeadBlock.Root()) {
```

Comment:
In path mode, if block N is not recoverable, N-1 wouldn't be either, right? Or is there a case where just the head few is missing (e.g. crash)?

Reply:
Wondering if we can be smarter about rewinds in path mode since we kind of know - in theory - the rewind point we can do. E.g. if we roll back prior to teh pivot (or the reorg threshold) then we should be able to know that and just jump to the genesis vs. iterating 17M blocks only to realise we found the genesis,